server/app/services/applicant/predicate/JsonPathPredicateGenerator.java

```diff
@@ -37,10 +41,39 @@ public JsonPathPredicateGenerator(
   /**
    * Formats a {@link LeafOperationExpressionNode} in JsonPath format: {@code path[?(expression)]}
    *
-   * <p>Example: \$.applicant.address[?(@.zip in ["12345", "56789"])]
+   * <p>Example: \$.applicant.name[?(@.last in ["Smith", "Lee"])]
    */
   public JsonPathPredicate fromLeafNode(LeafOperationExpressionNode node)
       throws InvalidPredicateException {
+    return JsonPathPredicate.create(
+        String.format(
+            "%s[?(@.%s %s %s)]",
+            getPath(node).predicateFormat(),
+            node.scalar().name().toLowerCase(),
+            node.operator().toJsonPathOperator(),
+            node.comparedValue().value()));
+  }
+
+  /**
+   * Formats a {@link services.program.predicate.LeafAddressServiceAreaExpressionNode} in JsonPath
+   * format: {@code path[?(expression)]}
+   *
+   * <p>Example: \$.applicant.address[?(@.service_area =~ /seattle_InArea_\d+/i)]
+   */
+  public JsonPathPredicate fromLeafAddressServiceAreaNode(LeafAddressServiceAreaExpressionNode node)
+      throws InvalidPredicateException {
+    return JsonPathPredicate.create(
+        String.format(
+            "%s[?(@.%s %s %s)]",
+            getPath(node).predicateFormat(),
+            Scalar.SERVICE_AREA.name().toLowerCase(),
+            Operator.IN_SERVICE_AREA.toJsonPathOperator(),
+            String.format(
+                "/([a-zA-Z\\-]+_[a-zA-Z]+_\\d+,)*%s_InArea_\\d+(,[a-zA-Z\\-]+_[a-zA-Z]+_\\d+)*/",
+                node.serviceAreaId())));
```

Comment:
I don't love this regex-based approach, but there are tradeoffs either way.

Pros:
- this code stays consistent for how it generally works between the two leaf node types
- this was really quick to implement!

Cons:
- regexes tend to be are write-only (illegible)
- does not differentiate between `NotInArea` and `Failed`

Reply:
If we add validation to the service area IDs, keeping the regex makes sense to me. And passing EsriServiceAreaState can make it more flexible with something like:
`String.format("/%1$s_(%2$s|%3$s)_(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d+Z)/", node.serviceAreaId, EsriServiceAreaState.IN_AREA, EsriServiceAreaState.FAILED)`


For reference, tested `([a-zA-Z]+)_(InArea|Failed)_(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d+Z)` against:
- Seattle_InArea_2023-02-07T11:49:22.456Z
- Bellevue_Failed_2023-02-07T11:49:22.4556666Z

Reply:
Cool, and good call on checking for `Failed` as well.

For the timestamps I recommend we use an integer unix timestamp since that's what we're doing elsewhere in the codebase.